net/src/headers.rs

```diff
@@ -73,6 +79,36 @@ impl Net {
             Net::Ipv6(ip) => ip.next_header(),
         }
     }
+
+    pub fn try_set_source(&mut self, addr: IpAddr) -> Result<(), NetError> {
```

Comment:
Should we update the util methods on packet that also do this to use these new methods instead of replicating the logic there?

Reply:
Also, we should make this take `UnicastIpAddr`

Reply:
Done, thank!